### DISCUSSION TITLE
How to disable _all_ warnings for a folder in my project

### DISCUSSION THREAD

**Discussion by intonarumori** [eclipse-cdt/cdt#891] [Q&A]
I searched high and low for a definitive answer on this, but I could not figure out how to suppress all warnings for a specific folder in my project. I've tried:
- Project Settings -> C/C++ Build -> Settings -> Turn off all warnings
- Project Settings -> C/C++ General -> Code Analysis -> Configure Workspace Settings -> Turn off all warnings
- Project Settings -> C/C++ General -> Code Analysis -> Configure Workspace Settings -> Select all warnings -> Exclude directory for all warnings

I still have warnings listed.

The warnings I have:
- Wimplicit-function-declaration
- Wattributes
- Wdiscarded-qualifiers
- comparision between pointer and integer

This is all in third party code I have to include in my project.

Any ideas what else to try?
I'm used to having zero warnings, turning warnings to errors in my own code and having a clean working environment.

**Comment by jld01** [comment]
The _Inhibit all warnings (-w)_ compiler option should do what you require:

![image](https://github.com/user-attachments/assets/3b313bf3-50c1-4e2f-853a-33627dac5fff)

Within CDT managed build projects, you can configure this option for a specific source folder containing your third-party code by right-clicking on the folder in the _Project Explorer_ view and selecting _Properties_ from the context menu.

If your project also contains C++ code, you may need to set this compiler option for both the C compiler and the C++ compiler.

Does this answer your question?

**Reply by arla03522** [reply]
I have restricted warning configuration for custom folders in the project: 
![image_2025-06-20_143336009](https://github.com/user-attachments/assets/4e5e9124-d29f-470f-9154-a89e33232702)
Any ideas?

Used eclipse version:
Eclipse IDE for Embedded C/C++ Developers (includes Incubating components)
Version: 2024-09 (4.33.0)
Build id: 20240905-0614

**Reply by arla03522** [reply]
Seems there are restrictions with "Arm Cross GCC" toolchain only. With "Cross GCC" toolchain all ok.

**Reply by jld01** [reply]
@arla03522, I suggest you raise this as an issue with the Eclipse Embedded CDT project: https://github.com/eclipse-embed-cdt/eclipse-plugins/issues

**Comment by gvl2023** [comment]
If you want to actually disable compiler warnings, as opposed to filtering them out from the Problems view, suggest you use compiler pragmas at the point of inclusion, you may want to group all problematic 3p headers into one header that disables the warnings before including them, and include that header in your code instead.

```C++
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wunused-parameter"
//add more of these as needed: #pragma GCC diagnostic ignored "-W..." 
#include "3pHeaderWithWarnings.h"
#pragma GCC diagnostic pop
```

If you simply want your warnings to not show up in the Problems View you can try to define a working set that excludes your 3p folder and have this mode selected:

![image](https://github.com/user-attachments/assets/40d1cb1f-c7ee-4e32-b7be-e85123023f09)
